eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/utils/loadresult/ErrorResult.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.utils.loadresult;
+
+import android.os.Parcel;
+
+
+/**
+ * Error {@link LoadResult}.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class ErrorResult<T> implements LoadResult<T>
+{
+    private final Exception mCauseException;
+
+
+    public ErrorResult(Exception causeException)
+    {
+        mCauseException = causeException;
+    }
+
+
+    @Override
+    public T result() throws Exception
```

Comment:
Since you have a specific Exception you should not throw a generic Exception here. Make it specific as well.